janitor/functions/filter.py

```diff
@@ -33,29 +34,50 @@ def filter_string(
 
         >>> import pandas as pd
         >>> import janitor
-        >>> df = pd.DataFrame({"a": range(3, 6), "b": ["bear", "peel", "sail"]})
+        >>> df = pd.DataFrame({"a": range(3, 6), "b": ["bear", "peeL", "sail"]})
         >>> df
            a     b
         0  3  bear
-        1  4  peel
+        1  4  peeL
         2  5  sail
         >>> df.filter_string(column_name="b", search_string="ee")
            a     b
-        1  4  peel
+        1  4  peeL
+        >>> df.filter_string(column_name="b", search_string="L", case=False)
+           a     b
+        1  4  peeL
+        2  5  sail
+
+    Example: Filter names don't contian `'.'` (disable regex mode).
+
+        >>> import pandas as pd
+        >>> import janitor
+        >>> df = pd.Series(["JoseChen", "Brian.Salvi"], name="Name").to_frame()
+        >>> df
+                  Name
+        0     JoseChen
+        1  Brian.Salvi
+        >>> df.filter_string(column_name="Name", search_string=".", regex=False, complement=True)
+               Name
+        0  JoseChen
 
     `search_string` is also permitted to be any valid regex pattern.
 
     :param df: A pandas DataFrame.
     :param column_name: The column to filter. The column should contain strings.
-    :param search_string: A regex pattern or a (sub-)string to search.
+    :param search_string: A regex pattern or a (sub-)string to search. It's also
+        `str.contains` parameter `pat`'s alias name.
     :param complement: Whether to return the complement of the filter or not. If
         set to True, then the rows for which the string search fails are retained
         instead.
+    :param kwargs: `str.contains`'s keyword arguements.
     :returns: A filtered pandas DataFrame.
     """  # noqa: E501
-    criteria = df[column_name].str.contains(search_string)
+
+    criteria = df[column_name].str.contains(search_string, **kwargs)
```

Comment:
since we have just a few kwargs for str.contains, I'd suggest listing them out explicitly; alternatively, maybe add a line, to refer the user to pd.Series.str.contains page.

Reply:
I don't know how MKDocs work to function link.
The following one work for Sphinx. And it could link to its doc.

```
:func:`pandas.Series.str.contains`
```

Reply:
There could add `See Also` section in doc or just use MKDocs function domain mark to link.

Reply:
If full `filter_string` all parameters,`search_string` should change to `pat` or not.
This is a problem.

Reply:
Have a look at the documentation for complete.py on how to link.  Let's wait for others to chime in as well

Reply:
> Have a look at the documentation for complete.py on how to link. Let's wait for others to chime in as well

I knew MKDocs link is Markdown link method via `[name](link)`.
So for there, ```[`pandas.Series.str.contains`](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.contains.html)```.
This doesn't look well.
MKDocs can't handle external-link like [sphinx.ext.intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#module-sphinx.ext.intersphinx).

Reply:
I'm in favour of `**kwargs`; it'll reduce our maintenance burden here. This probably should be a pattern that we follow, I guess.

Reply:
That said, I see that we've got explicit kwargs listed in there. Okie dokes - let's not hold up the PR anymore. I'm happy to approve.